lib/SystemStatistics.php

```diff
@@ -179,4 +184,24 @@ protected function getAppsInfo() {
 		return $info;
 	}
 
+	/**
+	 * Checks if a function is available. Borrowed from
+	 * https://github.com/nextcloud/server/blob/2e36069e24406455ad3f3998aa25e2a949d1402a/lib/private/legacy/helper.php#L475
+	 *
+	 * @param string $function_name
+	 * @return bool
+	 */
+	public function is_function_enabled($function_name) {
+		if (!function_exists($function_name)) {
+			return false;
+		}
+		if ($this->phpIni->listContains('disable_functions', $function_name)) {
+			return false;
+		}
+		if ($this->phpIni->listContains('suhosin.executor.func.blacklist', $function_name)) {
```

Comment:
Thank you! May i come up with another suggestion? I was wondering what happened to suhosin. Latest release is only available for php 5.6. There is another version for php 7.x (https://github.com/sektioneins/suhosin7) but they never released a production ready version. I guess there is only a little change that someone is using php7 with suhosin.

@nickvergessen do we need the check for suhosin.executor.func.blacklist?

Reply:
I was wondering about the exact same just right before when following your suggestion. According to most recent [System Requirements](https://docs.nextcloud.com/server/14/admin_manual/installation/system_requirements.html) of NC14 the runtime PHP 5.6 is still listed as supported, so I decided to keep the few lines for now.

However apps like `oauth2` or `theming` already takes advantage of the `Strict type declaration` which is available starting PHP 7 and up...

Reply:
fyi https://github.com/nextcloud/server/issues/7368

Reply:
NC14 requires php7+, the docs need adjustments

Im fine with leaving out the suhosin here

Reply:
OK. Removed then.